p2p/net/filter/filter.go

```diff
@@ -27,18 +26,24 @@ func (fs *Filters) AddDialFilter(f *net.IPNet) {
 }
 
 func (f *Filters) AddrBlocked(a ma.Multiaddr) bool {
-	_, addr, err := manet.DialArgs(a)
+	maddr := ma.Split(a)
+	if len(maddr) == 0 {
+		return false
+	}
+	netaddr, err := manet.ToNetAddr(maddr[0])
 	if err != nil {
 		// if we cant parse it, its probably not blocked
 		return false
 	}
+	netip := net.ParseIP(netaddr.String())
+	if netip == nil {
+		return false
```

Comment:
right now these filters are ip-specific. in the future we may want to express filters of other kinds (like tcp ports, etc)